ecl2df/compdat.py

```diff
@@ -173,14 +186,34 @@ def deck2dfs(deck, start_date=None, unroll=True):
 
     compsegs_df = pd.DataFrame(compsegsrecords)
     welsegs_df = pd.DataFrame(welsegsrecords)
+    wsegsicd_df = pd.DataFrame(wsegsicdrecords)
+    wsegaicd_df = pd.DataFrame(wsegaicdrecords)
 
     if unroll and not welsegs_df.empty:
         welsegs_df = unrolldf(welsegs_df, "SEGMENT1", "SEGMENT2")
 
+    if unroll and not wsegsicd_df.empty:
+        wsegsicd_df = unrolldf(wsegsicd_df, "SEGMENT1", "SEGMENT2")
+
+    if unroll and not wsegaicd_df.empty:
+        wsegaicd_df = unrolldf(wsegaicd_df, "SEGMENT1", "SEGMENT2")
+
     if "KEYWORD_IDX" in compdat_df.columns:
         compdat_df.drop(["KEYWORD_IDX"], axis=1, inplace=True)
 
-    return dict(COMPDAT=compdat_df, COMPSEGS=compsegs_df, WELSEGS=welsegs_df)
+    if "KEYWORD_IDX" in wsegsicd_df.columns:
+        wsegsicd_df.drop(["KEYWORD_IDX"], axis=1, inplace=True)
```

Comment:
I see this drop is not done for WELSEGS and COMPSEGS, for which WSEG*ICD are analogues. So potentially it should not be done for WSEG*ICD or it should also be done for WELSEGS and COMPSEGS?

Reply:
Hmm, her har det skjedd noe feil. La meg sjekke litt og gjøre en pull-request til.

Reply:
Well, WSEGxICD is analogous to COMPDAT with only one record, so I used COMPDAT as a template. WELSEGS and COMPSEGS have two records. KEYWORD_IDX is not defined in WELSEGS and COMPSEGS and therefore not dumped. If KEYWORD_IDX should have been defined/dumped for WELSEGS and COMPSEGS I think you should discuss with the developer for those keywords?

Reply:
Thanks, that sounds reasonable. Maybe that information should be included as a code comment.